Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -265,11 +280,41 @@ extension Processor {
     return true
   }
 
+  @inline(never)
+  @_effects(releasenone)
   func loadScalar() -> Unicode.Scalar? {
-    currentPosition < end ? input.unicodeScalars[currentPosition] : nil
+    guard currentPosition < end else { return nil }
+//    if let utf8 = self.fastUTF8 {
+//      let firstByte = utf8[currentPosition.encodedOffset]
+//      if firstByte < 0x80 {
+//        let returnValue = Unicode.Scalar(firstByte)
+//        // TODO: More comprehensive assertion framework to test before and after
+//        // TODO: unsafe-ish optimizations
+//        assert(returnValue == input.unicodeScalars[currentPosition])
+//
+//        return returnValue
+//      }
+//
+//    }
```

Comment:
TODO: investigate re-enabling this code, or even consider looking at all potential callers of `loadScalar`.